src/seedu/addressbook/commands/Command.java

```diff
@@ -39,12 +39,9 @@ public static String getMessageForPersonListShownSummary(List<? extends ReadOnly
     /**
      * Executes the command and returns the result.
      */
-    public CommandResult execute(){
-        throw new UnsupportedOperationException("This method should be implement in child classes");
-    }
+    public abstract CommandResult execute();
 
-    //Note: it is better to make the execute() method abstract, by replacing the above method with the line below:
-    //public abstract CommandResult execute();
+    public abstract boolean isMutating();
```

Comment:
This method should have a header comment. If not, users of the method as well as those who overrides method will not know how exactly the method behaves.

Take note that header comment specifies a contract so when you comment, use precise phrasing such as "Returns true if ..." instead of "Checks if ...".

An example of the header comment will be "Returns true if the Command **potentially** mutates the data". E.g. Adding a duplicate person will not mutate data but the corresponding `AddCommand` object will still return true for this method.